### PR TITLE
Documentation: update fleetd configuration medadata description

### LINKED ISSUE #1255: Please provide some more complex fleet scheduling examples
I'd like to be able to put in the fleet section of cloud-init:
    
    metadata: role=render role=ceph-osd

And I can.  It even appears in fleetctl list-machines.
However, when scheduling if you have a unit file containing

    [X-Fleet]
    MachineMetadata=role=render
    Global=true

And then submit and start that task, I'd expect ti to work wherever role=render is set.  However, it doesn't work presumably because role=ceph-osd is also set. 

Perhaps I can set role=render,ceph-osd
And use MachineMetadata=role=*render*.  But that may not always give the correct results if you're metadata names are not entirely distinct.

Most likely I'd have to instead go the route of

    metadata: render=true ceph-osd=true

And 

    [X-Fleet]
    MachineMetadata=render=true
    Global=true

### PR DESCRIPTION
**Pull request by wuqixuan**
Add some description about medadata on fleetd configuration.

Fixes #1255

### CODE REVIEW

File: Documentation/deployment-and-configuration.md [hunk 1]
```diff
@@ -115,6 +115,24 @@ Default: ""
 Comma-delimited key/value pairs that are published with the local to the fleet registry. This data can be used directly by a client of fleet to make scheduling decisions. An example set of metadata could look like:  
 
 	metadata="region=us-west,az=us-west-1"
+	metadata='region=us-west,az=us-west-1'
+	metadata=region=us-west,az=us-west-1
+
+Formally,the configuation of metadata should conform one of this three form:
```
Comment (bcwaldon) [inline Documentation/deployment-and-configuration.md]: The value of the metadata option should conform to one of these three forms:

File: Documentation/deployment-and-configuration.md [hunk 2]
```diff
@@ -115,6 +115,24 @@ Default: ""
...
+    
+	metadata="xxx"
+	metadata='xxx'
+	metadata=xxx
+
+while xxx is of the form:
```
Comment (bcwaldon) [inline Documentation/deployment-and-configuration.md]: ...while xxx is one of:

File: Documentation/deployment-and-configuration.md [hunk 3]
```diff
@@ -115,6 +115,24 @@ Default: ""
...
+
+	yyy[,yyy[,yyy...]]
+
+while yyy if of the form:
```
Comment (bcwaldon) [inline Documentation/deployment-and-configuration.md]: ... and yyy is one of:

File: Documentation/deployment-and-configuration.md [hunk 4]
```diff
@@ -115,6 +115,24 @@ Default: ""
...
+
+	key = value
+
+Space and tab are allow to appear beside '=' and ','. If give the same key in multiple pairs, the last value will take effect.
```
Comment (bcwaldon) [inline Documentation/deployment-and-configuration.md]: Space and tab characters are stripped after the equals sign and around each comma.
If the same key is defined more than once, the last value overwrites the previous value(s).

**Comment by wuqixuan** [comment]
Thanks for the comments, will update.

**Comment by wuqixuan** [comment]
Updated as per comments.

**Comment by wuqixuan** [comment]
@mischief @bcwaldon, do you have another comment for my latest modification?

File: Documentation/deployment-and-configuration.md [hunk 5]
```diff
@@ -115,6 +115,24 @@ Default: ""
...
+	metadata='region=us-west,az=us-west-1'
+	metadata=region=us-west,az=us-west-1
+
+The value of the metadata option should conform to one of these three forms:
+    
+	metadata="xxx"
+	metadata='xxx'
+	metadata=xxx
+
+...while xxx is one of:
```
Comment (jonboulle) [inline Documentation/deployment-and-configuration.md]: s/xxx/where/

File: Documentation/deployment-and-configuration.md [hunk 6]
```diff
@@ -115,6 +115,24 @@ Default: ""
...
+
+	yyy[,yyy[,yyy...]]
+
+...and yyy if one of:
```
Comment (jonboulle) [inline Documentation/deployment-and-configuration.md]: s/if one/is one/

File: Documentation/deployment-and-configuration.md [hunk 7]
```diff
@@ -115,6 +115,24 @@ Default: ""
...
+
+	key = value
```
Comment (jonboulle) [inline Documentation/deployment-and-configuration.md]: key=value

**Comment by wuqixuan** [comment]
@jonboulle ,updated as per comments.

**Comment by jonboulle** [comment]
thanks!

**Pull request merged**